backend/btrixcloud/operator/cronjobs.py

```diff
@@ -125,8 +125,20 @@ async def make_new_crawljob(
             )
             print("Scheduled Crawl Created: " + crawl_id)
 
-        profile_filename = await self.crawl_config_ops.get_profile_filename(
-            crawlconfig.profileid, org
+        profile_filename, profile_proxy_id = (
+            await self.crawl_config_ops.profiles.get_profile_filename_and_proxy(
+                crawlconfig.profileid, org
+            )
+        )
+        if crawlconfig.profileid and not profile_filename:
+            print(f"error: missing profile {crawlconfig.profileid}")
+            return self.get_finished_response(metadata)
+
+        # only save profile if selected proxy matches profile proxy
+        save_profile_id = (
+            str(crawlconfig.profileid)
+            if crawlconfig.profileid and crawlconfig.proxyId == profile_proxy_id
```

Comment:
Same comment as above, need to account for case where no proxy is set and so `crawlconfig.proxyId` may be `None` or an empty string, whereas profile_proxy_id will always be an empty string.